### PR TITLE
Save ticket and get ticket with id endpoints added

### PR DESCRIPTION
**Pull request by 93RakeshS**

### CODE REVIEW

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs [hunk 1]
```diff
@@ -8,17 +8,48 @@ namespace TicketBookingAPI.Controllers
...
 
         [HttpGet("/{pattern}")]
         public async Task<ActionResult<IEnumerable<CityModel>>> GetAllCities(string pattern)
         {
-            var cities = await _service.GetAllCitiesLike(pattern);
+            var cities = await _cityservice.GetAllCitiesLike(pattern);
             return Ok(cities);
         }
+
+        [HttpPost]
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs:28]: why it is a post method ?

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Extensions/RegisterDependenciesExtension.cs [hunk 2]
```diff
@@ -10,11 +10,10 @@ public static void AddDependencies(this WebApplicationBuilder builder)
         {
             builder.Services.AddScoped<CityRepository>();
             builder.Services.AddScoped<CityService>();
-
+            builder.Services.AddScoped<BusRepository>();
+            builder.Services.AddScoped<BusService>();
+            builder.Services.AddScoped<TicketService>();
+            builder.Services.AddScoped<TicketRepository>();
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Extensions/RegisterDependenciesExtension.cs:16]: FInd a better way to inject the dependencies, move it to extension method

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Program.cs [hunk 3]
```diff
@@ -21,6 +22,10 @@
     opt => opt.UseSqlServer(builder.Configuration.GetConnectionString("APIConnection"))
     .UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking));
 
+builder.Services.Configure<ApiBehaviorOptions>(options =>
+{
+    options.SuppressModelStateInvalidFilter = true;
+});
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Program.cs:28]: what is this code for ?

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        private IMapper _mapper;
+        public BusRepository(TicketManagemetContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+        public async Task<IEnumerable<BusModel>> GetBuses()
+        {
+
+            var buses = await _context.Bus.Include(bus => bus.DestinationCity).Include(bus => bus.SourceCity).ToListAsync();
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs:22]: Follow single line, singe dot rule

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    DestinationCity = bus.DestinationCity,
+                    BusName = bus.BusName,
+                    StartDateTime = bus.StartDateTime,
+                    EndDateTime = bus.EndDateTime,
+                    Type = bus.Type,
+                }).Where(bus =>
+                bus.StartDateTime == busSearchInput.StartDate 
+                && bus.SourceCity.Id == busSearchInput.SourceCityId
+                && bus.DestinationCity.Id == busSearchInput.DestinationCityId
+                ).ToListAsync();
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs:43]: First add the where condition then select

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/TicketRepository.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<TicketModel> SaveTicket(TicketModel ticketModel)
+        {
+            var ticketEntry = _mapper.Map<Ticket>(ticketModel);
+            _context.Entry(ticketEntry).State = EntityState.Added;
+            var res = _context.SaveChanges();
+            if(res > 0)
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/TicketRepository.cs:25]: No need of this condition

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/TicketRepository.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            {
+                var ticketRes = await _context.Ticket
+                    .Include(ticket => ticket.Bus)
+                    .ThenInclude(city => city.SourceCity)
+                    .Include(ticket => ticket.Bus)
+                    .ThenInclude(city => city.DestinationCity)
+                    .Where(ticket => ticket.Id == ticketEntry.Id).FirstOrDefaultAsync();
+                ticketModel = _mapper.Map<TicketModel>(ticketRes);
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/TicketRepository.cs:33]: First where condition then include, and why are you including these many entities ?

Comment (rahul-calvi) [reply TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/TicketRepository.cs:33]: you dont need to return the ticket, just return bool

**Review by rahul-calvi — CHANGES_REQUESTED** [review]

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs (hunk 1 above)
Comment (93RakeshS) [reply TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs:28]: we are sending request with model in body.

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs (hunk 5 above)
Comment (93RakeshS) [reply TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/BusRepository.cs]: done

**Pull request merged**